lib/cogctl/actions/users/info.ex

```diff
@@ -47,4 +39,18 @@ defmodule Cogctl.Actions.Users.Info do
     Enum.map(permissions, fn(perm) ->
       "#{perm["namespace"]}:#{perm["name"]}" end)
   end
+
+  defp get_roles_from_groups(groups) do
+    Enum.reduce(groups, [], fn(group, roles) ->
+                roles ++ get_roles(group.roles)
+    end)
+    |> Enum.uniq
+  end
+
+  defp get_roles(roles) do
```

Comment:
`get_roles_from_groups` and this `get_roles` function are still entangling too many operations. You need to first find the unique roles, then process them as table rows.